docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Scrambled Transaction to combat against MEV
+---
+
+### Program Statement
+
+The existence of MEV isn't ideal and costly.
+
+MEV motivates transaction reordering, censoring and copying. Also, it causes
+cluster destabilization because of incentivised forking.
+
+The root problem is that there is no rule in the pursuit of MEV frontier, except
+economic rationality. While this anarchism is welcomed, stemmed from the
+blockchain philosophy, it's also desirable for such an activity to economically
+impractical in the first place.
+
+Any temporal market imbalance should be corrected strictly on the grounds of a
+fair game, basically the golden first-come-first-served rule (with optional
+bidding for priority).
+
+Note that this proposal doesn't fully mitigate one of MEV types, unlike all
+other fully-neutralized types: back-running. Back-running can be exercised
+while the golden rule is strictly and fairly enforced.
+
+### Summary of Proposed Solution
+
+This proposal tries to combat against MEV, at the cost of (slight) global
+induced latency of cluster as a whole.
+
+At highest level, a full mitigation of MEV can be attained _if leaders (and
+searchers) can't know the contents of other's transactions until finishing to
+packing and ordering them into a new block_. In addition to this, the
+transaction details must also be deterministically revealed after the ordering
+of transactions is committed.
+
+So, naive 2-step transaction submission (encrypted tx, then its key) won't work,
+let alone increased latency.
+
+Instead, this proposal uses a special transaction construction _only to delay
+the revelation of its content for the short amount of time tied to confirmation
+by the cluster_. Specifically, transactions are encrypted by a private key
+derived from the recursive sha256, and its seed and cipher text are submitted to
+the cluster at the same time.
+
+This construction is called *scrambling* across this proposal.
+
+Conversely, the forced computation to reveal the transaction details by solving
+the recursive sha256 is called *descrambling*.
+
+### Summary of consensus changes
+
+In addition to the obvious change of submission of scrambled transactions by
+clients, this proposal introduces a new pipeline stage (called the descrambling
+stage) and splits the voting into 2 phases (pre-vote and vote).
+
+Firstly, the leader feeds incoming transactions down the turbine immediately
+with minimum sanitization (This assumes leaders are now so-called bank-less and
+scheduling is delayed and on-chain via mempool).
```

Comment:
Hi, based on my limited understanding currently Leader will validate each transaction (eg. it's signature and possibly that it isn't garbage and so on) **before** broadcasting it to the rest of network for inclusion in a block,

so with this "scrambling" approach you describe, would this validation be done **after** the transaction has already made it into block - in "descrambling" phase ? Meaning users can spam the network with garbage transactions that will be included in blocks but perhaps not executed due to being garbage, yet still consuming lots of bandwidth and storage space.

Reply:
@norwnd yeah, as you relied to yourself, ddos threat is same as before and after scrambling

Reply:
Thinking about it some more, I guess DoS is really a non-issue for this proposal overall because it also can be mitigated to large extent by, for example, limiting how many transactions single account can issue (per second) at network level - if such limiting doesn't yet exists and is desirable to make Solana more resilient overall.